pango.c

```diff
@@ -109,15 +113,21 @@ void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		*baseline = pango_layout_get_baseline(layout) / PANGO_SCALE;
 	}
 	g_object_unref(layout);
+
+	free(buf);
 }
 
 void pango_printf(cairo_t *cairo, const char *font,
 		double scale, bool markup, const char *fmt, ...) {
-	char buf[max_chars];
+	char *buf = malloc(max_chars);
```

Comment:
Follow-up pull request: use `vsnprintf(NULL, 0, fmt, args)` to allocate just what we need.

Reply:
Yeah, that seems like a much better idea.